testing/argsfilter/test_argsfilter_c.cpp

```diff
@@ -0,0 +1,395 @@
+
+    fpga_properties filter = NULL;
+    ASSERT_EQ(fpgaGetProperties(NULL, &filter), FPGA_OK);
+
+    strcpy(zero, "fpgainfo");
+    strcpy(one, "fme");
+    strcpy(two, "-F");
+    strcpy(three, function);
+    EXPECT_EQ(set_properties_from_args(filter, &result, &argc, argv), 0);
+    EXPECT_EQ(fpgaDestroyProperties(&filter), FPGA_OK);
+}
+
+/**
+ * @test       function_neg
+ * @brief      Test: set_properties_from_args
+ * @details    When passed with invalid argument for the function, <br>
+ *             the function returns an error. <br>
+ */
+TEST_P(argsfilter_c_p, function_neg) {
+    char zero[20];
+    char one[20];
+    char two[20];
+    char three[20];
+    char *argv[] = { zero, one, two, three };
+    int argc = 4;
+    fpga_result result;
+
+    fpga_properties filter = NULL;
+    ASSERT_EQ(fpgaGetProperties(NULL, &filter), FPGA_OK);
+
+    strcpy(zero, "fpgainfo");
+    strcpy(one, "fme");
+    strcpy(two, "-F");
+    strcpy(three, "zxyw");
+    EXPECT_NE(set_properties_from_args(filter, &result, &argc, argv), 0);
+    EXPECT_EQ(fpgaDestroyProperties(&filter), FPGA_OK);
+}
+
+/**
+ * @test       socket_id
+ * @brief      Test: set_properties_from_args
+ * @details    When passed with valid argument for the socket_id, <br>
+ *             the function returns 0. <br>
+ */
+TEST_P(argsfilter_c_p, socket_id) {
+    char zero[20];
+    char one[20];
+    char two[20];
+    char three[20];
+    char *argv[] = { zero, one, two, three };
+    int argc = 4;
+    fpga_result result;
+    char socket_id[2];
+
+    sprintf(socket_id, "0x%x", platform_.devices[0].socket_id);
+
+    fpga_properties filter = NULL;
+    ASSERT_EQ(fpgaGetProperties(NULL, &filter), FPGA_OK);
+
+    strcpy(zero, "fpgainfo");
+    strcpy(one, "fme");
+    strcpy(two, "-S");
+    strcpy(three, socket_id);
+    EXPECT_EQ(set_properties_from_args(filter, &result, &argc, argv), 0);
+    EXPECT_EQ(fpgaDestroyProperties(&filter), FPGA_OK);
+}
+
+/**
+ * @test       socket_neg
+ * @brief      Test: set_properties_from_args
+ * @details    When passed with invalid argument for the socket, <br>
+ *             the function returns an error. <br>
+ */
+TEST_P(argsfilter_c_p, socket_neg) {
+    char zero[20];
+    char one[20];
+    char two[20];
+    char three[20];
+    char *argv[] = { zero, one, two, three };
+    int argc = 4;
+    fpga_result result;
+
+    fpga_properties filter = NULL;
+    ASSERT_EQ(fpgaGetProperties(NULL, &filter), FPGA_OK);
+
+    strcpy(zero, "fpgainfo");
+    strcpy(one, "fme");
+    strcpy(two, "-S");
+    strcpy(three, "zxyw");
+    EXPECT_NE(set_properties_from_args(filter, &result, &argc, argv), 0);
+    EXPECT_EQ(fpgaDestroyProperties(&filter), FPGA_OK);
+}
+
+/**
+ * @test       segment
+ * @brief      Test: set_properties_from_args
+ * @details    When passed with valid argument for the segment, <br>
+ *             the function returns 0. <br>
+ */
+TEST_P(argsfilter_c_p, segment) {
```

Comment:
According to the coverage, the path for segment isn't being taken through the code.  I think it may require you to fully specify --segment.